plugins/cost-insights/src/utils/history.ts

```diff
@@ -15,9 +15,44 @@
  */
 
 import qs from 'qs';
-import { QueryParams } from '../types';
+import * as yup from 'yup';
+import { Group, PageFilters, getDefaultPageFilters, Duration } from '../types';
+import { ConfigContextProps } from '../hooks/useConfig';
 
-export const stringify = (queryParams: Partial<QueryParams>) =>
+const schema = yup
+  .object()
+  .shape({
+    group: yup.string(),
+    project: yup.string().nullable(),
+  })
+  .required();
+
+export const stringify = (queryParams: Partial<PageFilters>) =>
   qs.stringify(queryParams, { strictNullHandling: true });
-export const parse = (queryString: string): Partial<QueryParams> =>
+
+export const parse = (queryString: string): Partial<PageFilters> =>
   qs.parse(queryString, { ignoreQueryPrefix: true, strictNullHandling: true });
+
+export const validate = (queryString: string): Promise<PageFilters> => {
+  return schema.validate(parse(queryString), {
+    stripUnknown: true,
+    strict: true,
+  }) as Promise<PageFilters>;
+};
+
+export const getInitialPageState = (
+  groups: Group[],
+  queryParams: Partial<PageFilters> = {},
+) => {
+  return {
+    ...getDefaultPageFilters(groups),
+    ...(queryParams.project ? { project: queryParams.project } : {}),
+    ...(queryParams.group ? { group: queryParams.group } : {}),
+  };
+};
+
+export const getInitialProductState = (config: ConfigContextProps) =>
+  config.products.map(product => ({
+    productType: product.kind,
+    duration: Duration.P1M,
```

Comment:
Can you change this to P30D? We should default to current data when possible, as we do internally.